app/domain/authentication/authn_jwt/signing_key/create_jwks_from_http_response.rb

```diff
@@ -0,0 +1,53 @@
+module Authentication
+  module AuthnJwt
+    module SigningKey
+      # CreateJwksFromHttpResponse command class is responsible to create jwks object from http response
+      CreateJwksFromHttpResponse ||= CommandClass.new(
+        dependencies: {
+          logger: Rails.logger,
+          jwk_set_class: JSON::JWK::Set
+        },
+        inputs: %i[http_response]
+      ) do
+        def call
+          validate_response_exists
+          create_jwks_from_http_respnse
+        end
+
+        private
+
+        def validate_response_exists
+          raise Errors::Authentication::AuthnJwt::MissingHttpResponse if @http_response.blank?
+        end
+
+        def create_jwks_from_http_respnse
+          @logger.debug(LogMessages::Authentication::AuthnJwt::CreatingJwksFromHttpResponse.new)
+
+          raise Errors::Authentication::AuthnJwt::InvalidHttpResponseFormat unless @http_response.respond_to?(:body)
+
+          response_body = @http_response.body
+          encoded_body = Base64.encode64(response_body)
+          jwks, keys = parse_jwks_response(response_body, encoded_body)
+
+          if keys.blank?
+            raise Errors::Authentication::AuthnJwt::FetchJwksUriKeysNotFound, encoded_body
+          end
```

Comment:
sorry, i am still not sure why this check can't be done inside `parse_jwks_response`. then the method can return only `jwks` without returning `{ keys: @jwk_set_class.new(keys) }`

Reply:
If keys are blank we want one error and if there is any other failiure in the logic of parse_jwks_response we want another error.

What can be done is to use begin rescue block and after block inside of aprse_jwks_response to check for keys.blank?


WDYT

Reply:
i think that it is better. or even create another method named `verify_keys_are_non_empty` or something like this and call it inside `parse`